src/runtimes/node/bundlers/nft/index.ts

```diff
@@ -105,9 +106,9 @@ const traceFilesAndTranspile = async function ({
   runtimeAPIVersion: number
 }) {
   const isTypeScript = tsExtensions.has(extname(mainFile))
+  const tsFormat = isTypeScript ? getTSModuleFormat(mainFile) : MODULE_FORMAT.COMMONJS
```

Comment:
it's confusing to me that a variable called `tsFormat` has a case for when the file isn't `ts`. in the case of `!isTypeScript`, this value will be unused if i'm reading this right. does that make sense?

Reply:
It's because we're checking the value of `isTypeScript` twice: once here, and then again when it comes to returning the result (https://github.com/netlify/zip-it-and-ship-it/blob/220b0b03e0ec964635ed3245c206304bf6745022/src/runtimes/node/bundlers/nft/index.ts#L169). If we leave `tsFormat` as `undefined`, the compiler will complain that we're missing a value for `moduleFormat`, unaware that if we're in that branch we've already set a value for `tsFormat`.

The quick way around it was to have a fallback for `tsFormat` here, although we'll never use it.

If there's a more elegant way of doing it, I'm happy to change.